src/main/java/com/easytestit/generatexml/GenerateXML.java

```diff
@@ -1,145 +1,163 @@
-import org.apache.logging.log4j.Logger;
-import org.jetbrains.annotations.NotNull;
 
 import java.io.File;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collection;
-import java.util.Collections;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
 
 public class GenerateXML {
 
-    private static final Logger LOGGER = LogManager.getLogger(GenerateXML.class.getName());
-    private ConfigureXMLReport configureXMLReport;
+    private static final Logger LOGGER = Logger.getLogger(GenerateXML.class.getName());
+    private ConfigureXMLReport configuration;
 
+    /**
+     * Constructor a {@link GenerateXML()} class without parameters is necessary
+     * only if you need to use functionality with default parameters.
+     *
+     * Default parameter located in code in hardcode view.
+     */
     public GenerateXML() {
-        LOGGER.info("Starting application to convert JSON report to XML report with default parameters");
     }
 
-    public GenerateXML(@NotNull ConfigureXMLReport configureXMLReport) {
-        LOGGER.info("Starting application to convert JSON report to XML report with specified Configuration");
-        this.configureXMLReport = configureXMLReport;
+    /**
+     * Constructor a {@link GenerateXML()} class with {@link ConfigureXMLReport()} as a parameter
+     * is necessary when you need to use functionality with your specific configuration data.
+     *
+     * @param configuration the parameter which store specific configuration
+     */
+    public GenerateXML(final ConfigureXMLReport configuration) {
+        if (configuration != null)
+            this.configuration = configuration;
     }
 
     /**
-     * <p> Method describes functionality which provided files and generates the aggregated XML report.
+     * This method is the entry point to the application for converting files
+     * from JSON to XML. This method analyzes the configuration, which may have been passed
+     * to the constructor of class {@link GenerateXML()}.
+     *
+     * It is also analyzed depending on the configuration, what will be done.
+     * <p>For example, if the configuration was transferred without creating a
+     * ZIP of the archive, then this functionality {@link com.easytestit.generatexml.service.ZipService()#createZip()}
+     * for creating the archive will not be launched.
+     * <p/>
+     *
      * When generating process fails report with information about error is provided.
-     * </p>
      */
     public void make() {
         //create new folder where new XML report will be created
         createResultsReportFolder();
 
         try {
-            if (configureXMLReport != null) {
-                //read JSON files from compiled directory
-                configureXMLReport.addJsonFiles(getJsonFilesFrom(configureXMLReport.getReportFolder()));
+            if (configuration != null) {
+                if (configuration.getSource() == null) {
```

Comment:
if some method like getSource() can return no value make it **Optional**,
and after that use the next form:
`configuration.getSource().orElse(ConfigDataProvider.DEFAULT_FOLDER);`

Reply:
I was made global changes with the configuration at all. There isn't this row anymore.